lib/shared/install-kicad-library.ts

```diff
@@ -0,0 +1,149 @@
+import fs from "node:fs"
+import path from "node:path"
+import { glob } from "glob"
+
+/**
+ * Checks if a package name is a GitHub URL
+ */
+export function isGitHubUrl(packageName: string): boolean {
+  return (
+    packageName.startsWith("https://github.com/") ||
+    packageName.startsWith("http://github.com/") ||
+    packageName.startsWith("github:") ||
+    /^[\w-]+\/[\w-]+$/.test(packageName)
+  )
+}
+
+/**
+ * Extracts owner and repo from a GitHub URL or identifier
+ */
+export function extractGitHubInfo(packageName: string): {
+  owner: string
+  repo: string
+} | null {
+  // github:owner/repo format
+  if (packageName.startsWith("github:")) {
+    const [owner, repo] = packageName.replace("github:", "").split("/")
+    return { owner, repo }
+  }
+
+  // owner/repo format
+  if (/^[\w-]+\/[\w-]+$/.test(packageName)) {
+    const [owner, repo] = packageName.split("/")
+    return { owner, repo }
+  }
+
+  // Full GitHub URL
+  if (
+    packageName.startsWith("https://github.com/") ||
+    packageName.startsWith("http://github.com/")
+  ) {
+    const url = new URL(packageName)
+    const [, owner, repo] = url.pathname.split("/")
+    return { owner, repo: repo.replace(/\.git$/, "") }
+  }
+
+  return null
+}
+
+/**
+ * Generates TypeScript declarations for .kicad_mod files in types/ folder
+ */
+export async function generateKicadTypeDeclarations(
+  cwd: string = process.cwd(),
+): Promise<void> {
+  const nodeModulesPath = path.join(cwd, "node_modules")
+
+  if (!fs.existsSync(nodeModulesPath)) {
+    return
+  }
+
+  // Find all .kicad_mod files in node_modules
+  const kicadModFiles = await glob("**/*.kicad_mod", {
+    cwd: nodeModulesPath,
+    absolute: false,
+  })
+
+  if (kicadModFiles.length === 0) {
+    return
+  }
+
+  // Create types directory if it doesn't exist
+  const typesDir = path.join(cwd, "types")
+  if (!fs.existsSync(typesDir)) {
+    fs.mkdirSync(typesDir, { recursive: true })
+  }
+
+  // Generate type declaration file in types/ folder
+  const typeDeclarationPath = path.join(typesDir, "kicad_mod.d.ts")
+  const typeDeclaration = `declare module "*.kicad_mod" {
+  const value: string
```

Comment:
it loads as a circuit json array or any[] IIRC, you might be right but would appreciate double checking